src/pages/index.js

```diff
@@ -44,125 +54,220 @@ export const cardData = {
-const addFormElement = addCardModal.querySelector(".modal__form");
+api
+  .getInitialCards()
+  .then((cardData) => {
+    cardSection = new Section(
+      {
+        items: cardData.reverse(),
+        renderer: renderCard,
+      },
+      cardListSelector
+    );
+    cardSection.renderItems();
+  })
+  .catch((error) => {
+    console.error("Error loading initial cards:", error);
+  });
 
-const editFormValidator = new FormValidator(settings, editFormElement);
-const addFormValidator = new FormValidator(settings, addFormElement);
+// Likes Api
 
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
+function handleCardLikeClick(cardId, isLiked, setLikesStatus) {
+  if (isLiked) {
+    api
+      .unlikeCard(cardId)
+      .then((cardData) => {
+        setLikesStatus(cardData.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(cardId)
+      .then((cardData) => {
+        setLikesStatus(cardData.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
 
-// New Popupwithform const
+// Edit Avatar
 
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
+const editAvatarPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleEditAvatar,
+  "Save"
 );
-addCardPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
-editProfilePopup.setEventListeners();
-
-// Create and render card
+function handleEditAvatar(data) {
+  editAvatarPopup.setLoading(true);
+  api
+    .updateProfileAvatar(data.link)
+    .then((userData) => {
+      userInfo.setUserAvatar(userData.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarPopup.setLoading(false, "Saving");
+    });
+}
 
-const renderCard = (cardData) => {
-  const newCard = new Card(cardData, "#card-template", handleCardClick);
-  cardSection.addItem(newCard.getView());
-};
+editAvatarPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  cardListSelector
-);
+profileAvatarButton.addEventListener("click", () => {
+  avatarEditFormValidator.toggleButtonState();
 
-cardSection.renderItems();
+  editAvatarPopup.open();
+});
 
-//Preview Popup Const
+// User Info + Update Profile
 
-const imagePreviewPopup = new PopupWithImage("#preview-image-modal");
-imagePreviewPopup.setEventListeners();
+const userInfo = new UserInfo(profileTitle, profileDescription, avatarElement);
 
-function handleCardClick(name, link) {
-  imagePreviewPopup.open(name, link);
+function fillProfileForm() {
+  const userInfoData = userInfo.getUserInfo();
+  profileTitleInput.value = userInfoData.name;
+  profileDescriptionInput.value = userInfoData.description;
 }
 
-/*Event Handlers*/
+api
+  .getUserInfo()
```

Comment:
* Please, delete `api.getUserInfo()` from line 160. You've fetched the userData in line 57